src/components/ListItem/index.jsx

```diff
@@ -1,262 +1,88 @@
+import { useTheme } from "styled-components";
 import { StyledListItem, FlexContainer, PlayerInfo } from "./styledList";
 import * as G from "../";
-import { useTheme } from "styled-components";
-import { Button, InputPlayer, Typography } from "../";
+import { Button, InputPlayer } from "../";
+import formatText from "./formatText";
 import declineButton from "../../assets/icons/DeclineBtn.svg";
 import acceptButton from "../../assets/icons/AcceptBtn.svg";
 import deleteButton from "../../assets/icons/Deletar.svg";
 import editButton from "../../assets/icons/Editar.svg";
 
-//placeholder functions
-function editItem() {
-  console.log("aceito");
-}
-//id and set are props for type='player' and variation='edit'
-const ListItem = (props) => {
-  const {
-    children,
-    variation,
-    set,
-    id,
-    onClick,
-    type = "ranking",
-    ...restProps
-  } = props;
+const ListItem = ({
+  data,
+  id,
+  type = "ranking",
+  variation = "default",
+  onClick,
+  dispatch = () => {},
+  ...restProps
+}) => {
+  const styles = { type, variation };
   const theme = useTheme();
 
-  if (type === "player" && variation === "edit") {
-    function getDeleteItem(set, id) {
-      function deleteItem() {
-        set((players) => {
-          const temp = [...players];
-          temp.filter((player) => player.id !== id);
-          return temp;
-        });
-      }
-      return deleteItem;
-    }
-    const deleteItem = getDeleteItem(set, id);
-    return (
-      <FlexContainer key={id} type={type}>
-        <StyledListItem type={type}>
-          {children.map((item, index) =>
-            handleList(item, index, type, theme, id)
-          )}
-          <Button type="icon" onClick={editItem}>
-            <img src={editButton} alt="edit" width="20px" height="24px" />
+  const listComponents = {
+    player: {
+      edit: (
+        <FlexContainer key={id} type={type}>
+          <StyledListItem type={type}>
+            {formatText(data, styles, theme)}
+            <Button type="icon" onClick={() => {}}>
+              <img src={editButton} alt="edit" width="20px" height="24px" />
+            </Button>
+            <Button
+              type="icon"
+              onClick={() => {
+                dispatch({ type: "deletePlayer", payload: id });
+              }}
+            >
+              <img src={deleteButton} alt="delete" width="20px" height="24px" />
+            </Button>
+          </StyledListItem>
+        </FlexContainer>
+      ),
+      add: (
+        <PlayerInfo>
+          <InputPlayer type="number" />
+          <InputPlayer type="name" />
+          <Button type="icon">
```

Comment:
Eu fiquei pensando em como a gente vai pegar esses valores desses inputs, eu pensei de duas formas, mas as duas formas esbarram em refatorar esses componentes.
A gente poderia transformar esse PlayerInfo em form, e pegar os valores dos inputs no onSubmit só que pra isso a gente teria de passar o atributo type pra um dos botões pq o default é submit, um é submit só que o botão de cancelar não pode ser do tipo submit, se não quando a pessoa cancelar tem que receber o type button, só que o type button tá sendo usado como props daí teria de refatorar esse componente e em todos os lugares que usa ele.

A outra é transformar esse InputPlayer em um componente controlado, recebendo o estado e atualizando o estado quando mudar, eu acho que esse é mais viável de mudar pq eu acredito que aparece em menos lugares, o que que ces acham @teddiur @rednerdoficial

Reply:
Eu gostei da ideia de transformar em um componente que recebe estado porque daí isso já endereça o problema de quando o componente esta em [player][edit] e muda pra [player][add], a gente espera que os inputs estejam preenchidos com os valores antigos, certo?
Agora que eu to pensando, os nomes das variations estão ruins.